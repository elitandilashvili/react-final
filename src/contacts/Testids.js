export const TEST_IDS = {
  emailInput: {
    inputElement: 'emailInput',
    hasError: 'email.hasError',
    success: 'email.success',
  },

  passwordInput: {
    inputElement: 'passwordInput',
    hasError: 'password.hasError',
    success: 'password.success',
  },
};
