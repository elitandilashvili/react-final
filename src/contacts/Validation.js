export const MIN_PASSWORD_VALUE = 7;
export const MIN_EMAIL_VALUE = 12;
