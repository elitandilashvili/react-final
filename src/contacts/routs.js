export const HOME_PATH = '/';
export const LOGIN_PATH = '/login';
export const DOCUMENTATION_PATH = '/Private-page';
