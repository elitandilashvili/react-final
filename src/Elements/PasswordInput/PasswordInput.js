import { useState, useEffect } from 'react';
import PropTypes from 'prop-types';
import { MIN_PASSWORD_VALUE } from '../../contacts/Validation';
import { TEST_IDS } from '../../contacts/Testids';

export function PasswordInput({ name }) {
  const [value, setValue] = useState('');
  const [hasError, setHasError] = useState(false);

  useEffect(() => {
    if (value && value.length < MIN_PASSWORD_VALUE) {
      setHasError(true);
    } else if (value && value.length > MIN_PASSWORD_VALUE) {
      setHasError(false);
    }
  }, [value]);

  const renderClassName = () => {
    let className = 'input ';
    if (hasError && value) {
      className += 'is-danger';
    } else if (!hasError && value) {
      className += 'is-success';
    }

    return className;
  };

  return (
    <p className="control has-icons-left has-icons-right">
      <input
        className={renderClassName()}
        type="password"
        placeholder="Password"
        name={name}
        value={value}
        autoComplete="new-password"
        onChange={({ target }) => setValue(target.value)}
        required
        data-testid={TEST_IDS.passwordInput.inputElement}
      />
      <span className="icon is-small is-left">
        <i className="fas fa-lock"></i>
      </span>

      {!hasError && value && (
        <span
          className="icon is-small is-right"
          data-testid={TEST_IDS.passwordInput.success}
        >
          <i className="fas fa-check"></i>
        </span>
      )}

      {hasError && value && (
        <span
          className="icon is-small is-right"
          data-testid={TEST_IDS.passwordInput.hasError}
        >
          <i className="fas fa-exclamation-triangle" />
        </span>
      )}
    </p>
  );
}

PasswordInput.defaultProps = {
  name: 'password',
};

PasswordInput.propTypes = {
  name: PropTypes.string.isRequired,
};
