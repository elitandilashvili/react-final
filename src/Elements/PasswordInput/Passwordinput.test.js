import { render } from '@testing-library/react';
import userEvent from '@testing-library/user-event';
import { PasswordInput } from './PasswordInput';
import { TEST_IDS } from '../../contacts/Testids';

describe('PasswordInput', () => {
  it('should render correctly with correct props', () => {
    const testPassword = 'testPassword ';
    const Component = render(<PasswordInput name={testPassword} />);

    expect(
      Component.getByTestId(TEST_IDS.passwordInput.inputElement).getAttribute(
        'name'
      )
    ).toBe(testPassword);
  });

  it('should render error state', () => {
    const testPassword = 'testPassword';
    const Component = render(<PasswordInput name={testPassword} />);
    const inputElement = Component.getByTestId(
      TEST_IDS.passwordInput.inputElement
    );
    userEvent.type(inputElement, 'wrong');

    expect(
      Component.getByTestId(TEST_IDS.passwordInput.hasError)
    ).toBeInTheDocument();
  });

  it('should render success state', () => {
    const testPassword = ' testPassword';
    const Component = render(<PasswordInput name={testPassword} />);
    const inputElement = Component.getByTestId(
      TEST_IDS.passwordInput.inputElement
    );
    userEvent.type(inputElement, '123456789');

    expect(
      Component.getByTestId(TEST_IDS.passwordInput.success)
    ).toBeInTheDocument();
  });
});

describe('PasswordInput snapshots', () => {
  test('snap 1', () => {
    const testPassword = 'testPassword ';
    const Component = render(<PasswordInput name={testPassword} />);
    expect(Component).toMatchSnapshot();
  });
});

test('snap 2', () => {
  const testPassword = 'testPassword';
  const Component = render(<PasswordInput name={testPassword} />);
  const inputElement = Component.getByTestId(
    TEST_IDS.passwordInput.inputElement
  );
  userEvent.type(inputElement, 'wrong');

  expect(Component).toMatchSnapshot();
});

test('snap 3', () => {
  const testPassword = 'testPassword';
  const Component = render(<PasswordInput name={testPassword} />);
  const inputElement = Component.getByTestId(
    TEST_IDS.passwordInput.inputElement
  );
  userEvent.type(inputElement, '123456789');
  expect(Component).toMatchSnapshot();
});
