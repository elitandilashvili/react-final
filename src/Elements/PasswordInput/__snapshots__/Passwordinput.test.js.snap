// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PasswordInput snapshots snap 1 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <p
        class="control has-icons-left has-icons-right"
      >
        <input
          autocomplete="new-password"
          class="input "
          data-testid="passwordInput"
          name="testPassword "
          placeholder="Password"
          required=""
          type="password"
          value=""
        />
        <span
          class="icon is-small is-left"
        >
          <i
            class="fas fa-lock"
          />
        </span>
        
      </p>
    </div>
  </body>,
  "container": <div>
    <p
      class="control has-icons-left has-icons-right"
    >
      <input
        autocomplete="new-password"
        class="input "
        data-testid="passwordInput"
        name="testPassword "
        placeholder="Password"
        required=""
        type="password"
        value=""
      />
      <span
        class="icon is-small is-left"
      >
        <i
          class="fas fa-lock"
        />
      </span>
      
    </p>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`snap 2 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <p
        class="control has-icons-left has-icons-right"
      >
        <input
          autocomplete="new-password"
          class="input is-danger"
          data-testid="passwordInput"
          name="testPassword"
          placeholder="Password"
          required=""
          type="password"
          value="wrong"
        />
        <span
          class="icon is-small is-left"
        >
          <i
            class="fas fa-lock"
          />
        </span>
        <span
          class="icon is-small is-right"
          data-testid="password.hasError"
        >
          <i
            class="fas fa-exclamation-triangle"
          />
        </span>
      </p>
    </div>
  </body>,
  "container": <div>
    <p
      class="control has-icons-left has-icons-right"
    >
      <input
        autocomplete="new-password"
        class="input is-danger"
        data-testid="passwordInput"
        name="testPassword"
        placeholder="Password"
        required=""
        type="password"
        value="wrong"
      />
      <span
        class="icon is-small is-left"
      >
        <i
          class="fas fa-lock"
        />
      </span>
      <span
        class="icon is-small is-right"
        data-testid="password.hasError"
      >
        <i
          class="fas fa-exclamation-triangle"
        />
      </span>
    </p>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`snap 3 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <p
        class="control has-icons-left has-icons-right"
      >
        <input
          autocomplete="new-password"
          class="input is-success"
          data-testid="passwordInput"
          name="testPassword"
          placeholder="Password"
          required=""
          type="password"
          value="123456789"
        />
        <span
          class="icon is-small is-left"
        >
          <i
            class="fas fa-lock"
          />
        </span>
        <span
          class="icon is-small is-right"
          data-testid="password.success"
        >
          <i
            class="fas fa-check"
          />
        </span>
      </p>
    </div>
  </body>,
  "container": <div>
    <p
      class="control has-icons-left has-icons-right"
    >
      <input
        autocomplete="new-password"
        class="input is-success"
        data-testid="passwordInput"
        name="testPassword"
        placeholder="Password"
        required=""
        type="password"
        value="123456789"
      />
      <span
        class="icon is-small is-left"
      >
        <i
          class="fas fa-lock"
        />
      </span>
      <span
        class="icon is-small is-right"
        data-testid="password.success"
      >
        <i
          class="fas fa-check"
        />
      </span>
    </p>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
